src/main/resources/com/google/api/codegen/java/sample.snip

```diff
@@ -9,10 +9,12 @@
    *    instructions at https://github.com/google/google-api-java-client.
    *    On other build systems, you can add the jar files to your project from
    *    https://developers.google.com/resources/api-libraries/download/{@sampleClass.apiName}/{@sampleClass.apiVersion}/java
-   * 3. This sample uses Application Default Credentials for authentication.
-   *    If not already done, install the gcloud CLI from
-   *    https://cloud.google.com/sdk and run
-   *    `gcloud beta auth application-default login`
+   @if sampleClass.body.authType == "APPLICATION_DEFAULT_CREDENTIALS"
```

Comment:
Though it may make the snippet file messier, I still think it makes more sense for this non-language-specific pre-req to go before the client library pre-req.